bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/BaseModbusThingHandler.java

```diff
@@ -43,23 +42,53 @@
 public abstract class BaseModbusThingHandler extends BaseThingHandler {
     private List<PollTask> periodicPollers = Collections.synchronizedList(new ArrayList<>());
     private List<Future<?>> oneTimePollers = Collections.synchronizedList(new ArrayList<>());
-    private volatile boolean initialized;
 
     public BaseModbusThingHandler(Thing thing) {
         super(thing);
     }
 
     /**
-     * This method must be invoked in the base class' initialize() method before any other initialization is done.
-     * It will throw an unchecked exception if the {@link ModbusCommunicationInterface} is not accessible (fail-fast).
-     * This prevents any further initialization of the Thing. The framework will set the ThingStatus to
-     * HANDLER_INITIALIZING_ERROR and display the exception's message.
+     * This method is called when the Thing is being initialized, but only if the Modbus Bridge is configured correctly.
+     * The code that normally goes into `BaseThingHandler.initialize()` like configuration reading and validation goes
+     * here.
      */
+    public abstract void modbusInitialize();
+
     @Override
-    public void initialize() {
-        getModbus();
+    final public void initialize() {
+        try {
+            // check if the Bridge is configured correctly (fail-fast)
+            getModbus();
+            getBridgeHandler().getSlaveId();
+
+            modbusInitialize();
+        } catch (EndpointNotInitializedException | IllegalStateException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
```

Comment:
BRIDGE_OFFLINE would be more correct for cases when bridge is offline? It would direct the user to look at the bridge error message.

Are there any other reasons we expect the initialization to fail?

Reply:
I saw there's a ThingStatusDetail.BRIDGE_UNINITIALIZED. I think that would fit even better. I moved the invocation of the abstract method out of the try-block, that only a Bridge error could lead to a BRIDGE_UNINITIALIZED.